### PR TITLE
Setup Backbone testing with mocha, chai, sinon

### LINKED ISSUE #213: Setup backbone testing framework

### PR DESCRIPTION
**Pull request by Trinovantes**
The tests are headless so they need to load JS libs via require (loaded
from npm repository). As a result, we are no longer using libraries
loaded from bower. Those loaded from bower cannot be (easily) tested.

I also updated our package.json to remove some warnings.

Closes #213

### CODE REVIEW

**Comment by ConnorCimowsky** [comment]
:thumbs:

File: src-backbone/tests/smoke_test.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+var CreateSuite = require('utils/create_suite');
+
+CreateSuite('Smoke Test', function() {
+    describe('Array#indexOf', function () {
+        it('should return -1 when the value is not present', function () {
+            assert.equal(-1, [1, 2, 3].indexOf(4));
```
Comment (Ommy) [inline src-backbone/tests/smoke_test.js]: Why?

**Comment by Ommy** [comment]
LGTM once comment is addressed

File: src-backbone/tests/smoke_test.js (hunk 1 above)
Comment (Trinovantes) [inline src-backbone/tests/smoke_test.js]: To make sure our tests can run at all without depending on our main app. ¯\_(ツ)_/¯

**Pull request merged**